File: practice/algorithms/implementation/bon_appetit/bon_appetit.py
#!/usr/bin/env python

n, k = map(int, raw_input().strip().split())
c = map(int, raw_input().strip().split())
b = int(raw_input().strip())

total = sum([c[POS] for POS in xrange(n) if POS != k])
if 2 * b <= total:
    print "Bon Appetit"
else:
    print b - total // 2

File: practice/algorithms/implementation/bon_appetit/bon_appetit.cpp
#include <iostream>

using namespace std;

int main()
{
    long n, k, b, total = 0, c;
    cin >> n >> k;
    for (int pos = 0; pos < n; ++pos)
    {
        cin >> c;
        if (pos != k)
        {
            total += c;
        }
    }
    
    cin >> b;
    
    if (total >= 2 * b)
    {
        cout << "Bon Appetit" << endl;
    }
    else
    {
        cout << (b - total / 2) << endl;
    }
    
    return 0;
}

